### ISSUE TITLE
Enhancement: Improve generation quality with extended base model training

### ISSUE THREAD

**Issue by DrDrewCain** [DrDrewCain/MiniMacLLM#9]
## Problem

Current generation quality is incoherent due to undertrained base model:
- Only 5 epochs on WikiText-2
- Model hasn't converged (loss still decreasing)
- Outputs show poor grammar and topic drift

Example output:
```
Prompt: "Python is a programming language that"
Output: "Python is a programming language that be in , , in , , . the of is , an ated , , , and..."
```

## Goal

Achieve coherent, grammatically correct generation from the base model before continual learning.

### 1. Extended Training (Recommended)
Train base model for 20-50 epochs instead of 5:
```bash
python scripts/pretrain.py \
  --config configs/medium.yaml \
  --data data/raw/wikitext2_train.txt \
  --tokenizer data/tokenizers/wikitext_8k \
  --epochs 30 \
  --batch_size 4 \
  --max_seq_len 1024 \
  --save_dir checkpoints/wikitext_medium_v2
```

Expected improvements:
- Lower perplexity (< 20)
- Coherent sentence structure
- Better context retention

### 2. Training Monitoring
Add convergence criteria:
- Track validation loss
- Save best checkpoint (not just final)
- Early stopping when loss plateaus
- Learning rate scheduling (cosine decay)

### 3. Quality Metrics
Implement automatic quality checks:
- Perplexity on validation set
- Token accuracy
- Generation diversity (distinct n-grams)
- Human eval on sample outputs

## Acceptance Criteria

- [ ] Base model trained to convergence (validation loss plateau)
- [ ] Perplexity < 20 on WikiText-2 validation
- [ ] Generated text is grammatically coherent
- [ ] Model completes sentences logically
- [ ] Validation loss tracked during training

## Resources Needed

- **Compute:** M3 Max can handle this (~2-4 hours for 30 epochs)
- **Data:** WikiText-2 sufficient for now
- **Metrics:** Add perplexity calculation to training loop

## Related

- Blocked by: None
- Blocks: Quality demonstration of continual learning
- Related to: #9 (more data), #10 (hyperparameter tuning)

---

**Priority:** High - Foundation for all downstream tasks  
**Difficulty:** Easy - Just more training time  
**Impact:** Critical - Makes results presentable
